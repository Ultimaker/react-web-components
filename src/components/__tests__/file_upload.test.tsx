// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import FileUpload from '../file_upload';

describe('The FileUpload component', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            id: 'testFileUpload',
            onChangeHandler: jest.fn(),
        };
        wrapper = shallow(<FileUpload {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
        expect(props.onChangeHandler).not.toHaveBeenCalled();
    });

});
