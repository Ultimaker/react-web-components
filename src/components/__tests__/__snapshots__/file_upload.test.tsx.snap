// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The FileUpload component should render 1`] = `
<div
  className="file-upload"
  onClick={[Function]}
>
  <input
    id="testFileUpload"
    name="testFileUpload"
    onChange={[Function]}
    placeholder="A text placeholder"
    type="file"
  />
  <div
    className="layout layout--gutter-sm"
  >
    <div
      className="layout__item u-fill file-upload__input-container"
    >
      <label
        className="file-upload__input"
        htmlFor="testFileUpload"
      />
    </div>
    <div
      className="layout__item u-fit"
    >
      <label
        className="btn btn--primary"
        htmlFor="testFileUpload"
      >
        Choose file
      </label>
    </div>
  </div>
</div>
`;
